lib/pure/os.nim

```diff
@@ -17,33 +17,701 @@
 include "system/inclrtl"
 
 import
-  strutils, times
+  strutils
 
-when defined(windows):
-  import winlean
+when defined(nimscript):
+  discard
+elif defined(windows):
+  import winlean, times
 elif defined(posix):
-  import posix
+  import posix, times
 
   proc toTime(ts: Timespec): times.Time {.inline.} =
     result = initTime(ts.tv_sec.int64, ts.tv_nsec.int)
-
 else:
   {.error: "OS module not ported to your operating system!".}
 
-import ospaths
-export ospaths
+when defined(nimscript) and defined(nimErrorProcCanHaveBody):
```

Comment:
/cc @Araq @dom96 
should that be:
```nim
when defined(nimscript):
  when defined(nimErrorProcCanHaveBody):
    {.pragma: noNimScript, error: "this proc is not available on the NimScript target".}
  else:
    {.pragma: noNimScript, error.}
else:
  {.pragma: noNimScript.}
```

Reply:
No, ``nimErrorProcCanHaveBody`` means the Nim compiler allows .error procs to have a body that is then not sem'checked. Very useful here.